scripts/helpers.py

```diff
@@ -565,3 +566,62 @@ def locate_bus(
         return gdf_co[gdf_co.geometry == min(gdf_co.geometry, key=(point.distance))][
             col
         ].item()  # looks for closest one shape=node
+
+
+def get_last_commit_message():
+    """
+    Function to get the last Git commit message
+    Returns
+    -------
+    result : string
+    """
+    try:
+        # Run the Git command to get the last commit message
+        result = subprocess.run(
+            ["git", "log", "-1", "--pretty=format:%H %s"],
+            capture_output=True,
+            text=True,
+        )
+        return result.stdout.strip()
+    except Exception as e:
+        logging.warning(f"Error getting the last commit message: {e}")
+        return ""
+
+
+def get_submodule_commit_message():
+    """
+    Function to get the last PyPSA-Earth Git commit message
+    Returns
+    -------
+    result : string
+    """
+    try:
+        # Retrieve the last commit message for pypsa-earth
+        submodule_path = f"pypsa-earth"
+        last_commit_message = (
+            subprocess.check_output(
+                ["git", "log", "-n", "1", "--pretty=format:%H %s"],
+                cwd=submodule_path,
+                stderr=subprocess.STDOUT,
+            )
+            .decode()
+            .strip()
+        )
+        return last_commit_message
+    except subprocess.CalledProcessError as e:
+        logging.warning(f"Error executing Git: {e}")
+        return None
```

Comment:
I see some duplication here, may we create a general function such as `get_last_commit_message(pdir=".")` where pdir represents the git folder we desire to get the commit of?
In case of the repository, that function may be "." and in case of the submodule, it may be "./pypsa-earth/", or something like that.
By doing that, we would need a single function for both.